deepmd/pt/utils/auto_batch_size.py

```diff
@@ -78,26 +104,44 @@ def execute_with_batch_size(
             )
 
         index = 0
-        results = []
+        results = None
         while index < total_size:
             n_batch, result = self.execute(execute_with_batch_size, index, natoms)
-            if not isinstance(result, tuple):
-                result = (result,)
+            if not self.returned_dict:
+                result = (result,) if not isinstance(result, tuple) else result
             index += n_batch
-            if n_batch:
-                for rr in result:
-                    rr.reshape((n_batch, -1))
-                results.append(result)
-        r_list = []
-        for r in zip(*results):
+
+            def append_to_list(res_list, res):
+                if n_batch:
+                    res_list.append(res)
```

Comment:
no need to reshape?

Reply:
no the shape of the returned tensors are preserved. A concatenation is performed on the first (batch size) dimension of batch calculated tensors. plz see 
https://github.com/deepmodeling/deepmd-kit/blob/3c4ff2c1c34675c93b853902fd6b0d699034479a/deepmd/pt/utils/auto_batch_size.py#L132-L139